### ISSUE TITLE
Accessing property directly on a task result makes Scriban unable to find the specified property

### ISSUE THREAD

**Issue by zlepper** [scriban/scriban#382]
```c#
    public class ValueWrapper
    {
        public string Value { get; set; }


        public ValueWrapper(string value)
        {
            Value = value;
        }
    }
    
    public class Tests
    {

        [Test]
        public async Task AccessDirectlyOnFunctionResult()
        {
            var templateBody = "{{my_function().value}}";

            var templateContext = new TemplateContext
            {
                EnableRelaxedMemberAccess = false,
                StrictVariables = true
            };

            var template = Template.Parse(templateBody);
            Assert.That(template.HasErrors, Is.False);

            var so = new ScriptObject();
            so.Import("my_function", new Func<Task<ValueWrapper>>(async () =>
            {
                await Task.Delay(1);
                return new ValueWrapper("hello");
            }));
            
            templateContext.PushGlobal(so);

            var result = await template.RenderAsync(templateContext);
            
            Assert.That(result, Is.EqualTo("hello"));
        }
        
        [Test]
        public async Task IndirectAccess()
        {
            var templateBody = @"{{v = my_function()
v.value}}";

            var templateContext = new TemplateContext
            {
                EnableRelaxedMemberAccess = false,
                StrictVariables = true
            };

            var template = Template.Parse(templateBody);
            Assert.That(template.HasErrors, Is.False);

            var so = new ScriptObject();
            so.Import("my_function", new Func<Task<ValueWrapper>>(async () =>
            {
                await Task.Delay(1);
                return new ValueWrapper("hello");
            }));
            
            templateContext.PushGlobal(so);

            var result = await template.RenderAsync(templateContext);
            
            Assert.That(result, Is.EqualTo("hello"));
        }
    }
```

`AccessDirectlyOnFunctionResult` fails with `<input>(1,17) : error : Cannot get member with name value.`
`IndirectAccess` works fine. 

I would expect the first test to work also. I imagine it's the problem with the result being a task, and Scriban somehow not awaiting it when it's not being assigned to a local first.

**Comment by JvanderStad** [comment]
There are more issues with async/await methods.
When using parentheses the resulting `Task` doesn't get awaited.

See the following example

**example.linq** ([LINQPad](https://www.linqpad.net/))
```c#
<Query Kind="Program">
  <NuGetReference>Scriban</NuGetReference>
  <Namespace>Scriban</Namespace>
  <Namespace>System.Threading.Tasks</Namespace>
  <Namespace>Scriban.Runtime</Namespace>
  <Namespace>System.Text.Json</Namespace>
</Query>

#nullable enable

public static class Sample
{
	public class TaskResult
	{
		public string Member { get; set; } = "Hello Scriban";
	}
	public static Task<TaskResult> AsyncTask()
	{
		return Task.FromResult<TaskResult>(new ());
	}
	public static TaskResult SyncTask()
	{
		return new();
	}
	public static string Serialize(object? input)
	{
		return JsonSerializer.Serialize(input);
	}
}

public async Task RunAsync(string script, bool ok, string? message = null)
{
	script.Trim().Dump("Script");
	// SET UP
	var context = new TemplateContext();

	var global = new ScriptObject();
	global.Import(typeof(Sample));

	context.PushGlobal(global);

	// RUN
	(await Template.Parse(script, lexerOptions: new()
	{
		Mode = Scriban.Parsing.ScriptMode.ScriptOnly
	})
	.RenderAsync(context))
	.Dump($"Output");
	
	$"{ (ok ? "✔" : "⚠")} {message}".Dump("Result");
}

async Task Main()
{
	Util.RawHtml("<h1>Direct call</h1>").Dump();

	await RunAsync(@"sync_task | serialize", true);
	
	Util.RawHtml("<hr/>").Dump();

	await RunAsync(@"async_task | serialize", true);
	

	// ------------------
	Util.RawHtml("<h1>Parentheses</h1>").Dump();
	
	await RunAsync(@"
		output = (sync_task)
		output | serialize
	", true);

	Util.RawHtml("<hr/>").Dump();

	// BUG:
	await RunAsync(@"
		output = (async_task)
		output | serialize
	", false, "Task is not awaited");
}
```
**Output**
![image](https://user-images.githubusercontent.com/5426370/145662967-006b1464-2a1f-425e-ba6e-deab151dc667.png)

**Comment by xoofx** [comment]
This should be fixed by commit 88526a7, a bit more annoying to fix than expected.
